lib/utils/nodeFieldIndices.mjs

```diff
@@ -0,0 +1,118 @@
+import getAtRuleParams from './getAtRuleParams.mjs';
+import getRuleSelector from './getRuleSelector.mjs';
+
+import { isObject, isString } from './validateTypes.mjs';
+
+/** @import {AtRule, Declaration, Rule} from 'postcss' */
+
+/**
+ * @param {AtRule} atRule
+ * @returns {number}
+ */
+export function atRuleParamIndex(atRule) {
+	let index = atRuleRawsAfterNameIndex(atRule);
+
+	if (atRule.raws.afterName) {
+		index += atRule.raws.afterName.length;
+	}
+
+	return index;
+}
+
+/**
+ * @param {AtRule} node
+ * @returns {number}
+ */
+export function atRuleRawsAfterIndex(node) {
+	// subtract 1 for `}`
+
+	if (!node.source?.end?.offset) return node.toString().length - 1;
+
+	if (!node.raws?.after?.length) return node.source.end.offset - 1;
+
+	return node.source.end.offset - (node.raws.after.length + 1);
```

Comment:
[suggestion] To make the code more readable and reduce the evaluation count, how about extracting to local variables like this?

```suggestion
	const endOffset = node.source?.end?.offset;

	if (!endOffset) return node.toString().length - 1;

	const afterLength = node.raws?.after?.length;

	if (!afterLength) return endOffset - 1;

	return endOffset - (afterLength + 1);
```